client/src/CalendarEntry.jsx

```diff
@@ -1,14 +1,21 @@
 import React from 'react';
 import PropTypes from 'prop-types'
-import { Wrapper, Day, Date, Month } from './styles/CalendarEntry-style'
+import { RegularWrapper, SelectedWrapper, Day, Date, Month } from './styles/CalendarEntry-style'
 
-const CalendarEntry = ({ date }) => (
-  <Wrapper>
-    <Day className="day">{date.day}</Day>
-    <Date className="date">{date.date}</Date>
-    <Month className="month">{date.month}</Month>
-  </Wrapper>
-);
+const CalendarEntry = ({ date, clickFn, selected }) => {
+  return selected ?
+    <SelectedWrapper onClick={() => { clickFn(date); }}>
+      <Day className="day">{date.day}</Day>
+      <Date className="date">{date.date}</Date>
+      <Month className="month">{date.month}</Month>
+    </SelectedWrapper> :
+
+    <RegularWrapper onClick={() => { clickFn(date); }}>
+      <Day className="day">{date.day}</Day>
+      <Date className="date">{date.date}</Date>
+      <Month className="month">{date.month}</Month>
+    </RegularWrapper>
+};
```

Comment:
If the only difference is CSS a cleaner approach would be to pass the selected prop down to the styled component and trigger the change there.  You should be able to do the ternary statement there on the actual props which would result in slightly less code and an easier ability to trace the differences.

Reply:
i will refactor this after the component as a whole seems finished. Thanks!